### PR TITLE
Fix windows kubectl log -f.

### LINKED ISSUE #81735: kubectl logs -f doesn't work for Windows node
`kubectl logs -f` doesn't work for Windows node.

After some investigation, I think the problem is that for Docker the log file `kubectl logs` reads is a symlink to the actual docker container log file.

On linux, I believe fsnotify follows symlink by default https://github.com/fsnotify/fsnotify/issues/199, so it works on linux.

However, on windows I believe it doesn't follow the symlink. I tried to replace the symlink file with an actual file and append log to it, `kubectl logs -f` will start working after that.

The proposed solution is to explicitly follow the symlink in Kubelet, so that we get more consist behavior across different platforms.

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/kubernetes/kubernetes/issues/81735.

fsnotify doesn't have consistent behavior for symlinks. We explicitly follow the symlink to avoid the inconsistency between platforms.

I've verified this fix in my cluster, and it fixed the issue.

```release-note
Fix `kubectl logs -f` for windows server containers.
```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/logs/logs.go [hunk 1]
```diff
@@ -271,6 +272,15 @@ func (w *logWriter) write(msg *logMessage) error {
...
 // just pass in empty string "".
 func ReadLogs(ctx context.Context, path, containerID string, opts *LogOptions, runtimeService internalapi.RuntimeService, stdout, stderr io.Writer) error {
+	// fsnotify has different behavior for symlinks in different platform,
+	// for example it follows symlink on Linux, but not on Windows,
+	// so we explicitly resolve symlinks before reading the logs.
+	// There shouldn't be security issue because the container log
+	// path is owned by kubelet and the container runtime.
+	path, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return fmt.Errorf("failed to try resolving symlinks in path %q: %v", path, err)
```
Comment (dashpole) [inline pkg/kubelet/kuberuntime/logs/logs.go]: Since you are re-using the path variable, won't path always be empty string when err != nil?

Comment (Random-Liu) [reply pkg/kubelet/kuberuntime/logs/logs.go]: True. Will update.

Comment (Random-Liu) [reply pkg/kubelet/kuberuntime/logs/logs.go]: Done

**Comment by yujuhong** [comment]
Thanks! Add a release note so we can cherry-pick?

@kubernetes/sig-windows-bugs

**Comment by Random-Liu** [comment]
@yujuhong DOne

**Review by michmike — APPROVED** [review]
/approve
/lgtm

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Pull request merged**